src/pharmpy/model/external/nonmem/update.py

```diff
@@ -1482,6 +1482,44 @@ def update_abbr_record(model: Model, rv_trans):
     return model, trans
 
 
+def add_efim_records(control_stream, idx_cov, last_est_rec):
+    data_record = str(control_stream.get_records("DATA")[0]).rstrip()
+    input_record = str(control_stream.get_records("INPUT")[0]).rstrip()
+    efim_records = [
+        "$PROBLEM DESIGN",
+        f"{data_record} REWIND",
+        f"{input_record}",
+        "$MSFI efim.msf",
+        "$DESIGN APPROX=FO FIMDIAG=1 GROUPSIZE=1 OFVTYPE=1",
+    ]
+
+    for i, rec in enumerate(efim_records):
```

Comment:
If you do `enumerate(efim_records, 1)` it will start at 1 and will not have to be added in `insert_record`!

Reply:
Very true! Great catch 🎣. Force pushed.